### PR TITLE
TYP: nargsort

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Some simplification available if im right about #52764

### CODE REVIEW

File: pandas/core/sorting.py [hunk 1]
```diff
@@ -425,16 +465,27 @@ def nargsort(
         )
 
     if isinstance(items, ABCRangeIndex):
-        return items.argsort(ascending=ascending)  # TODO: test coverage with key?
+        return items.argsort(ascending=ascending)
     elif not isinstance(items, ABCMultiIndex):
         items = extract_array(items)
+    else:
+        raise TypeError(
```
Comment (mroeschke) [inline pandas/core/sorting.py:472]: Do we have a test that hits this?

Comment (jbrockmendel) [reply pandas/core/sorting.py:472]: No

Comment (jbrockmendel) [reply pandas/core/sorting.py:472]: this isn't public and the place in Index that calls nargsort specifically goes for a separate path for MultiIndex

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**